### PR TITLE
Mode manager prevents redundant mode changes

### PR DESCRIPTION
**Pull request by norro**
Before executing a requested mode change, i.e. applying the according changes, it is now checking whether the mode is already active. If so, no action is taken, but mode change service request is still returned with success.
    
https://github.com/micro-ROS/system_modes/issues/60

### CODE REVIEW

File: system_modes/test/launchtest/redundant_mode_changes.launch.py.in [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            ament_index_python.packages.get_package_share_directory(
+                'system_modes') + '/launch/mode_manager.launch.py'),
+        launch_arguments={'modelfile': modelfile}.items())
+
+    test_nodes = ExecuteProcess(
+        cmd=[
+            "@PYTHON_EXECUTABLE@",
+            "@TEST_NODES@"
+            ],
+        name='test_two_lifecycle_nodes',
```
Comment (nielsvd) [inline system_modes/test/launchtest/redundant_mode_changes.launch.py.in:39]: Copy-paste leftover? Should perhaps be `test_redundant_mode_changes`?

**Review by nielsvd — CHANGES_REQUESTED** [review]
Check one open comment.

File: system_modes/test/launchtest/redundant_mode_changes.launch.py.in (hunk 1 above)
Comment (norro) [reply system_modes/test/launchtest/redundant_mode_changes.launch.py.in:39]: This should be fine. This only names the process/executable that is part of the test - and it is indeed just two lifecycle nodes for testing.

**Review by nielsvd — APPROVED** [review]

**Comment by norro** [comment]
Thanks for the review, Niels!

**Pull request merged**